### ISSUE TITLE
Raft commit index is not monotonic

### ISSUE THREAD

**Issue by tangruize** [bakwc/PySyncObj#166]
## Describe the bug

According to the [raft paper](https://raft.github.io/raft.pdf),  commitIndex should increase monotonically. But under certain conditions pysyncobj violates it.

The root cause is that `self.__raftCommitIndex` is set to `min(leaderCommitIndex, self.__getCurrentLogIndex())` unconditionally when processing msg `append_entries`.

https://github.com/bakwc/PySyncObj/blob/d9c2deb7c0403ac26c1cfdcf6e29c9dc0a44e1e4/pysyncobj/syncobj.py#L921

## To Reproduce

1. Init a cluster with 3 nodes.
2. Node 1 becomes Leader and send `append_entries` to other nodes. And its log is `[{"cmd":"NO_OP","idx":1,"term":0},{"cmd":"NO_OP","idx":2,"term":1}]`
3. Node 2 and Node 3 receive the `append_entries`, and replies `next_node_idx 2` 
4. Node 1 receives the `next_node_idx 2`.
5. Node 1 sends `append_entries`
6. Node 2 and Node 3 receive the `append_entries`, and replies `next_node_idx 3`
7. Node 1 receives the `next_node_idx 3` and updates its `__raftMatchIndex` of other nodes to `2`.
8. Node 1 advances `__raftCommitIndex` to `2`.
9. Node 2 election times out, becomes leader, and sends `append_entries` with `__raftCommitIndex=1`.
10. Node 1 receives the `append_entries`, and sets `__raftCommitIndex=1`, which violates the monotonicity specification.

## A Possible Fix

Add a condition check:

```python3
if leaderCommitIndex > self.__raftCommitIndex:
    self.__raftCommitIndex = min(leaderCommitIndex, self.__getCurrentLogIndex())
```

**Comment by bakwc** [comment]
Was able to stably reproduce it in the random test (added a small delay before sending packets). I will create a new release after fixing a similar bug with match idx.

**Comment by bakwc** [comment]
Fixed in #168
